test/upgrade/prober/wathola/event/services.go

```diff
@@ -135,13 +159,25 @@ func (s *stepStore) reportProgress() {
 	}
 }
 
-func (e *ErrorStore) throw(format string, args ...interface{}) {
+func (e *ErrorStore) throwDuplicated(format string, args ...interface{}) {
+	e.duplicatedThrown = e.throwHelper(e.duplicatedThrown, format, args...)
+}
+
+func (e *ErrorStore) throwMissing(format string, args ...interface{}) {
+	e.missingThrown = e.throwHelper(e.missingThrown, format, args...)
+}
+
+func (e *ErrorStore) throwUnexpected(format string, args ...interface{}) {
+	e.unexpectedThrown = e.throwHelper(e.unexpectedThrown, format, args...)
+}
+
+func (e *ErrorStore) throwHelper(errThrown []thrown, format string, args ...interface{}) []thrown {
```

Comment:
Same here. Use meaningful name, rather than `throwHelper` (especially as it is almost identical to the other `thrownHelper` method).

I think cleaner code would be to have a method: `newThrown(format string, args []interface{}) thrown` used directly in those concrete methods above. This method shouldn't be in `(e *ErrorStore)` scope of course.

```golang
func (e *ErrorStore) throwUnexpected(format string, args ...interface{}) {
  e.unexpectedThrown = append(e.unexpectedThrown, newThrown(format, args))
}

func newThrown(format string, args []interface{}) thrown {
  return thrown{
    format: format,
    args:   args,
  }
}
```

Reply:
I changed the name to `appendThrown` instead because:
1. the `newThrown` if not really useful here as we can easily use the struct directly in this case.
2. we do have logging inside this function, so the reason I want this helper function is to have this kind of common code.

Reply:
Ok.